Scripts/altapay_config.php

```diff
@@ -0,0 +1,212 @@
+{
+
+    public function run()
+    {
+        $apiUser = "username";
+        $apiPass = "password";
+        $url     = "shopurl";
+
+        try {
+            $api      = new TestAuthentication(new Authentication($apiUser, $apiPass, $url));
+            $response = $api->call();
+            if (!$response) {
+                echo "API credentials are incorrect";
+                exit();
+            }
+        } catch (ClientException $e) {
+            echo "Error:" . $e->getMessage();
+            exit();
+        } catch (Exception $e) {
+            echo "Error:" . $e->getMessage();
+            exit();
+        }
+
+        $terminals = array();
+
+        try {
+            $api      = new Terminals(new Authentication($apiUser, $apiPass, $url));
+            $response = $api->call();
+            foreach ($response->Terminals as $terminal) {
+                $terminals[] = $terminal->Title;
+            }
+        } catch (ClientException $e) {
+            echo "Error:" . $e->getMessage();
+            exit();
+        } catch (Exception $e) {
+            echo "Error:" . $e->getMessage();
+            exit();
+        }
+
+        // Save API details
+
+        $this->resourceConfig->saveConfig(
+            'payment/altapay_config/api_log_in',
+            $apiUser,
+            'default',
+            0
+        );
+
+        $this->resourceConfig->saveConfig(
+            'payment/altapay_config/api_pass_word',
+            $this->encryptor->encrypt($apiPass),
+            'default',
+            0
+        );
+
+        $this->resourceConfig->saveConfig(
+            'payment/altapay_config/productionurl',
+            $url,
+            'default',
+            0
+        );
+
+        // Save Order Status
+
+        $this->resourceConfig->saveConfig(
+            'payment/altapay_status/before',
+            'pending',
+            'default',
+            0
+        );
+
+        $this->resourceConfig->saveConfig(
+            'payment/altapay_status/process',
+            'processing',
+            'default',
+            0
+        );
+
+        $this->resourceConfig->saveConfig(
+            'payment/altapay_status/fraud',
+            'fraud',
+            'default',
+            0
+        );
+
+        $this->resourceConfig->saveConfig(
+            'payment/altapay_status/cancel',
+            'canceled',
+            'default',
+            0
+        );
+
+        $this->resourceConfig->saveConfig(
+            'payment/altapay_status/autocapture',
+            'processing',
+            'default',
+            0
+        );
+
+        for ($i = 1; $i <= 5; $i++) {
+
+            $this->resourceConfig->saveConfig(
+                'payment/terminal' . $i . '/active',
+                1,
+                'default',
+                0
+            );
+
+            $this->resourceConfig->saveConfig(
+                'payment/terminal' . $i . '/title',
+                $terminals[$i],
```

Comment:
This does not seem to be generic. What happens if the terminal list is more or less than 5 terminals?

Reply:
In Magento 2 we can only set 5 terminals from the configuration that's why we put this loop in here.

Reply:
@BushraAsif what if the number of terminals available is only 4? This will fail

Reply:
@emicha, This has been updated.